src/libraries/pretest.proj

```diff
@@ -68,11 +68,31 @@
              ContinueOnError="ErrorAndContinue" />
   </Target>
 
+  <Target Name="RunCrossgen2OnHost" Condition="'$(Crossgen2Host)' == 'true'">
+    <ItemGroup>
+      <FilesToCrossgen Include="$(TestHostRuntimePath)\Microsoft.*.dll" Exclude="$(TestHostRuntimePath)\Microsoft.CodeAnalysis*.dll;$(TestHostRuntimePath)\Microsoft.DiaSymReader.Native.*.dll" />
+      <FilesToCrossgen Include="$(TestHostRuntimePath)\System.*.dll" Exclude="$(TestHostRuntimePath)\System.Runtime.WindowsRuntime.dll" />
+    </ItemGroup>
+
+    <MakeDir Directories="$(TestHostRuntimePath)\crossgen2out" />
+    <Exec Command="$(TestHostRootPath)\dotnet $(TestHostRuntimePath)\crossgen2.dll %(FilesToCrossgen.FullPath) --targetarch=x64 -O --inputbubble -r:$(TestHostRuntimePath)\System.*.dll -r:$(TestHostRuntimePath)\Microsoft.*.dll -r:$(TestHostRuntimePath)\mscorlib.dll -r:$(TestHostRuntimePath)\netstandard.dll -o:$(TestHostRuntimePath)\crossgen2out\%(FilesToCrossgen.Filename)%(FilesToCrossgen.Extension)" 
```

Comment:
This is running on the shared framework that it is modifying at the same time.  Typically we do not use the built product to perform build operations.  Should you use the stable build framework under .dotnet instead?

This command line probably needs quotes around paths.  Also, can crossgen2 just accept a folder for -r?  Seems fragile to require named references when it could just probe for them.  Probing seems simpler than resolving the wildcard in the app which it seems to be doing now.